### ISSUE TITLE
fetchGML_IDs should filter first with a POST, and then extract the GLM IDS

### ISSUE THREAD

**Issue by jread-usgs** [USGS-R/geoknife#277]
the current implementation fails when a feature has more than 5000 values (or polys)

**Comment by jread-usgs** [comment]
see https://github.com/USGS-R/hydroMaps/blob/master/R/plotWSB.R#L97

**Comment by jread-usgs** [comment]
```XML
<?xml version=\"1.0\"?><wfs:GetFeature xmlns:wfs=\"http://www.opengis.net/wfs\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns:gml=\"http://www.opengis.net/gml\" service=\"WFS\" version=\"1.1.0\" outputFormat=\"shape-zip\" xsi:schemaLocation=\"http://www.opengis.net/wfs http://schemas.opengis.net/wfs/1.1.0/wfs.xsd\"><wfs:Query xmlns:feature=\"http://owi.usgs.gov/NWC\" typeName=\"feature:epa_basins\" srsName=\"EPSG:4326\"><ogc:Filter xmlns:ogc=\"http://www.opengis.net/ogc\"><ogc:Or><ogc:PropertyIsEqualTo  matchCase=\"true\"><ogc:PropertyName>site_no</ogc:PropertyName><ogc:Literal>01573000</ogc:Literal></ogc:PropertyIsEqualTo><ogc:PropertyIsEqualTo  matchCase=\"true\"><ogc:PropertyName>site_no</ogc:PropertyName><ogc:Literal>01576000</ogc:Literal></ogc:PropertyIsEqualTo></ogc:Or></ogc:Filter></wfs:Query></wfs:GetFeature>
```
where attribute and value are needed in there.

**Comment by jread-usgs** [comment]
@dblodgett-usgs if I PR this in the future, you willing to review?

**Comment by dblodgett-usgs** [comment]
For sure!

**Comment by jread-usgs** [comment]
```XML
<?xml version=\"1.0\"?>
<wfs:GetFeature xmlns:wfs=\"http://www.opengis.net/wfs\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns:gml=\"http://www.opengis.net/gml\" service=\"WFS\" version=\"1.1.0\" outputFormat=\"shape-zip\" xsi:schemaLocation=\"http://www.opengis.net/wfs http://schemas.opengis.net/wfs/1.1.0/wfs.xsd\">
  <wfs:Query xmlns:feature=\"http://owi.usgs.gov/NWC\" typeName=\"derivative:CONUS_States\" propertyName=\"STATE\">
    <ogc:Filter xmlns:ogc=\"http://www.opengis.net/ogc\">
      <ogc:Or>
        <ogc:PropertyIsEqualTo  matchCase=\"true\">
          <ogc:PropertyName>STATE</ogc:PropertyName>
          <ogc:Literal>West Virginia</ogc:Literal>
        </ogc:PropertyIsEqualTo>
        <ogc:PropertyIsEqualTo  matchCase=\"true\">
          <ogc:PropertyName>STATE</ogc:PropertyName>
          <ogc:Literal>Wyoming</ogc:Literal>
        </ogc:PropertyIsEqualTo>
      </ogc:Or>
    </ogc:Filter>
  </wfs:Query>
</wfs:GetFeature>
```
working on this with a POST, but this still returns features, instead of the attribute values for `STATE` which is what I would get with the `GET` (no filtering): `http://cida.usgs.gov/gdp/geoserver/wfs?service=WFS&version=1.1.0&request=GetFeature&typename=derivative:CONUS_States&MAXFEATURES=10000&propertyname=STATE`

hmm...

**Comment by jread-usgs** [comment]
Looks like I need to specify the attribute name to be returned:
```XML
<?xml version=\"1.0\"?>
<wfs:GetFeature xmlns:ogc="http://www.opengis.net/ogc" xmlns:wfs=\"http://www.opengis.net/wfs\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns:gml=\"http://www.opengis.net/gml\" service=\"WFS\" version=\"1.1.0\" xsi:schemaLocation=\"http://www.opengis.net/wfs http://schemas.opengis.net/wfs/1.1.0/wfs.xsd\">
  <wfs:Query typeName=\"derivative:CONUS_States\">
    <ogc:PropertyName>STATE</ogc:PropertyName>
    <ogc:Filter xmlns:ogc=\"http://www.opengis.net/ogc\">
      <ogc:Or>
        <ogc:PropertyIsEqualTo  matchCase=\"true\">
          <ogc:PropertyName>STATE</ogc:PropertyName>
          <ogc:Literal>West Virginia</ogc:Literal>
        </ogc:PropertyIsEqualTo>
        <ogc:PropertyIsEqualTo  matchCase=\"true\">
          <ogc:PropertyName>STATE</ogc:PropertyName>
          <ogc:Literal>Wyoming</ogc:Literal>
        </ogc:PropertyIsEqualTo>
      </ogc:Or>
    </ogc:Filter>
  </wfs:Query>
</wfs:GetFeature>
``` 
returns what I want, which is:
```
<?xml version="1.0" encoding="UTF-8"?>
<wfs:FeatureCollection xmlns:waters="gov.usgs.cida.gdp.waters" xmlns:ogc="http://www.opengis.net/ogc" xmlns:draw="gov.usgs.cida.gdp.draw" xmlns:wfs="http://www.opengis.net/wfs" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:sample="gov.usgs.cida.gdp.sample" xmlns:ows="http://www.opengis.net/ows" xmlns:gml="http://www.opengis.net/gml" xmlns:derivative="gov.usgs.cida.gdp.derivative" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:upload="gov.usgs.cida.gdp.upload" numberOfFeatures="2" timeStamp="2016-06-09T01:56:22.213Z" xsi:schemaLocation="http://www.opengis.net/wfs http://cida.usgs.gov/gdp/geoserver/schemas/wfs/1.1.0/wfs.xsd gov.usgs.cida.gdp.derivative http://cida.usgs.gov/gdp/geoserver/wfs?service=WFS&amp;version=1.1.0&amp;request=DescribeFeatureType&amp;typeName=derivative%3ACONUS_States">
  <gml:featureMembers>
    <derivative:CONUS_States gml:id="CONUS_States.47">
      <derivative:STATE>West Virginia</derivative:STATE>
    </derivative:CONUS_States>
    <derivative:CONUS_States gml:id="CONUS_States.49">
      <derivative:STATE>Wyoming</derivative:STATE>
    </derivative:CONUS_States>
  </gml:featureMembers>
</wfs:FeatureCollection>
```

so, just need to implement that programmatically

**Comment by dblodgett-usgs** [comment]
Looks like you got things all figured for PR #286 here. These look pretty solid.

Something to note though, you don't actually have to do the gmlid lookup if you have implemented filtering like this. The GDP can take the filtered WFS request rather than a list of gmlids. You do have to include `<wfs:PropertyName>the_geom</wfs:PropertyName>` for geoserver-shapefile WFS and possibly other names for the geometry column for other WFS implementations, but you could skip the gmlid lookup part.

**Comment by jread-usgs** [comment]
Fixed in #286
